core/src/main/scala/org/ensime/core/Project.scala

```diff
@@ -55,6 +55,20 @@ class Project(
   }
   private val classfileWatcher = context.actorOf(Props(new ClassfileWatcher(config, searchService :: reTypecheck :: Nil)), "classFileWatcher")
 
+  var seenWorkSinceIdle = true
+  //  var lastMessageReceived = System.currentTimeMillis()
+
+  def processIdleMessage(): Unit = {
+    if (seenWorkSinceIdle)
+      seenWorkSinceIdle = false
+    else
+      System.gc()
```

Comment:
you need to run finalizers as well, see https://github.com/fommil/sbt-big-project/blob/master/src/main/scala/BigProjectSettings.scala#L106-L108